apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/create/components/client/create-ecosystem-form.client.tsx

```diff
@@ -0,0 +1,183 @@
+"use client";
+
+import { ConfirmationDialog } from "@/components/ui/ConfirmationDialog";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { ImageUpload } from "@/components/ui/image-upload";
+import { Input } from "@/components/ui/input";
+import { RadioGroup, RadioGroupItem } from "@/components/ui/radio-group";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2 } from "lucide-react";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import { z } from "zod";
+import { useCreateEcosystem } from "../../hooks/use-create-ecosystem";
+
+const formSchema = z.object({
+  name: z.string().min(1, {
+    message: "Ecosystem name is required",
+  }),
+  logo: z.instanceof(File, {
+    message: "Logo is required",
+  }),
+  permissions: z.union([z.literal("PARTNER_WHITELIST"), z.literal("ANYONE")]),
+});
+
+const nameToSlug = (name = "") => `${name.toLowerCase().replaceAll(" ", "-")}`;
+
+export function CreateEcosystemForm() {
+  const [confirmationOpen, setConfirmationOpen] = useState(false);
+  const router = useRouter();
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      permissions: "PARTNER_WHITELIST",
+    },
+  });
+
+  const { createEcosystem, isLoading } = useCreateEcosystem({
+    onError: (error) => {
+      const message =
+        error instanceof Error ? error.message : "Failed to create ecosystem";
+      toast.error(message);
+    },
+    onSuccess: (id: string) => {
+      form.reset();
+      router.push(`/dashboard/connect/ecosystem/${id}`);
+    },
+  });
+
+  return (
+    <>
+      <Form {...form}>
+        <form
+          onSubmit={form.handleSubmit(() => setConfirmationOpen(true))}
```

Comment:
this is a _little_ odd in terms of how the state flows.

I think a slightly better way of achieving this could be:
1. on form submission take the data (passed to the `handleSubmit` callback)
2. set it into a "formDataToBeConfirmed" state
3. if hat stat exists, show the confirmation modal with the data *in* the state
4. use the data from the state to create the ecosystem and re-set the state to undefined / null

This avoids the problem of "the form state is accessed twice", _generally_ you likely want to avoid any `form.getValues()` calls because they usually mean that you are pulling state from a weird direction instead of simply reacting to form submissions.